tokio/src/task/blocking.rs

```diff
@@ -34,7 +34,7 @@ cfg_rt_threaded! {
     {
         use crate::runtime::{enter, thread_pool};
 
-        enter::exit(|| thread_pool::block_in_place(f))
+        enter::try_exit(|| thread_pool::block_in_place(f)).0
```

Comment:
Maybe `enter::exit` should be moved **into** `thread_pool`. It could then conditionally be applied depending on whether or not there is a core.

Reply:
8cbf066487677fa21186a2772c81d157f03f014c